File: Single.java
class Base //parent
{
    public int A,B;// char

    public Base()//default constructor
    {
        System.out.println("inside Base deafult constructor");
   
    }
    public void fun(){System.out.println("inside base fun");}

}

class Derived extends Base //derived is child
{
    public int P,Q;//cshar
    public Derived(){
        System.out.println("inside Derived deafult constructor");
    }
    public void gun(){System.out.println("inside base gun");}
}

public class Single {
    public static void main(String[] args) {
        Derived dobj = new Derived(); //no casting
        System.out.println("value of Base A"+dobj.A);
        dobj.fun();
        dobj.gun();
    }
}

File: CPP/single.cpp
#include<iostream>
using namespace std;
class Demo //parent class
{
    public:
        int x,y;

        Demo()
        {
           cout<<"1. inside demo constructore"<<endl;
           x = 10;
           y = 20;
        }
        
        ~Demo()
        {
           cout<<"1. inside demo destructore"<<endl;
        }

        void fun()
        {
            cout<<"inside fun of Demo "<<endl;
        }
};

//child class
//
class Hello : public Demo 
{
    public:
        int a,b,c;
    Hello()
    {
        cout<<"2.inside Hello constructor "<<endl;
        a = 30;
        b = 40;
        c = 50;
    }
    ~Hello()
    {
        cout<<"2.inside Hello destructor"<<endl;
    }

    void gun()
    {
        cout<<"inside gun of Hello"<<endl;
    }

};

int main()
{
    Hello hobj;

    cout<<sizeof(hobj)<<"bytes"<<endl; //20 bytes

    cout<<hobj.x<<endl;
    cout<<hobj.y<<endl;
    cout<<hobj.a<<endl;
    cout<<hobj.b<<endl;
    cout<<hobj.c<<endl;

    hobj.fun();
    hobj.gun();


    return 0;

}